tests/unit/test_strike_service.py

```diff
@@ -0,0 +1,178 @@
+
+    mocked_exile_user = mocker.patch("moddingway.services.exile_service.exile_user")
+
+    res = await strike_service._apply_punishment(
+        mocked_embed, mocked_user, mocked_db_user, previous_points
+    )
+    if total_points >= 15:
+        mocked_ban.assert_called_once_with(
+            mocked_user,
+            "Your strike were severe or frequent to be removed from NA Ultimate Raiding - FFXIV",
+            False,
+        )
+    else:
+        mocked_ban.assert_not_called()
+        spy_calculate_punishment.assert_called_once_with(previous_points, total_points)
+        punishment_days = spy_calculate_punishment.spy_return
+        if punishment_days == 0:
+            mocked_exile_user.assert_not_called()
+        else:
+            mocked_exile_user.assert_called_once_with(
+                mocked_embed,
+                mocked_user,
+                mocker.ANY,
+                "Your actions were severe or frequent enough for you to receive this exile",
+            )
+    assert res == expected_punishment
+
+
+@pytest.mark.parametrize(
+    "user_id,total_points,strike_params,expected_result",
+    [
+        (
+            "1",
+            "1",
+            [("1", enums.StrikeSeverity.MINOR, "test", "1")],
+            "Strikes found for <@1>: [Temporary points: None | Permanent points: None]"
+            "\n* ID: 1 | SEVERITY: 1 | Moderator: <@1> | REASON: test"
+            "\nTotal Points: 1",
+        ),
+        (
+            "1",
+            "3",
+            [
+                ("1", enums.StrikeSeverity.MINOR, "test", "1"),
+                ("2", enums.StrikeSeverity.MODERATE, "test2", "3"),
+            ],
+            "Strikes found for <@1>: [Temporary points: None | Permanent points: None]"
+            "\n* ID: 1 | SEVERITY: 1 | Moderator: <@1> | REASON: test"
+            "\n* ID: 2 | SEVERITY: 2 | Moderator: <@3> | REASON: test2"
+            "\nTotal Points: 3",
+        ),
+    ],
+)
+@pytest.mark.asyncio
+async def test_get_user_strikes(
+    user_id: str,
+    total_points: str,
+    strike_params: List[tuple],
+    expected_result: str,
+    mocker: MockerFixture,
+    create_db_user,
+):
+    mocked_user = mocker.Mock(id=user_id)
+    mocked_db_user = create_db_user(get_strike_points=total_points)
+    mocker.patch(
+        "moddingway.database.users_database.get_user", return_value=mocked_db_user
+    )
+    mocker.patch(
+        "moddingway.database.strikes_database.list_strikes", return_value=strike_params
+    )
+    res = await strike_service.get_user_strikes(mocked_user)
+    assert res == expected_result
+
+
+@pytest.mark.asyncio
+async def test_add_strike(create_db_user, mocker: MockerFixture):
+
+    mocked_db_user = create_db_user(user_id=1, temporary_points=0, get_strike_points=0)
+
+    mocked_user = mocker.Mock(spec=discord.Member, id=1)
+
+    mocked_author = mocker.Mock(spec=discord.Member)
+    mocked_author.id = 2
+
+    mocked_logging_embed = mocker.Mock(spec=discord.Embed)
```

Comment:
Mocked embeds might be a good candidate for a fixture as well, if you want. Not a blocker though